solutions/platinum/usaco-817.mdx

```diff
@@ -0,0 +1,141 @@
+---
+id: usaco-817
+source: USACO Platinum 2018 February
+title: New Barns
+author: Justin Ji
+---
+
+<Spoiler title = "Hint 1">
+
+The furthest distance from a node is always to one endpoint of a diameter of
+the tree. After adding a node, how can we maintain the diameter of its tree?
+
+</Spoiler>
+
+<Spoiler title = "Answer to Hint 1">
+
+Consider using the diameter before adding this node to find the diameter after
+adding the node in question.
+
+</Spoiler>
+
+<Spoiler title = "Explanation">
+
+Consider answering all the queries online. As given in Hint 1, the furthest
+distance from a given node is to an endpoint of a diameter of our current tree.
+Now the problem boils down to keeping track of the diameter of each tree inside
+our graph after we add a node.
+
+Observe that if we add a leaf to the tree that becomes an endpoint of our
+new diameter, then the other endpoint of our new diameter is one of the endpoints
+of our old diameter. To understand why this is the case, consider laying out
+the tree in the following manner:
+
+![](<usaco-817/Diameter.png>)
```

Comment:
make the file name lowercase
maybe some alt text

Reply:
done